File: src/services/tauri/discord.ts
import { callCommand } from "@/services/tauri/client";

const DISCORD_CLIENT_ID = "1485730113550028902";

export type DiscordRpcActivityInput = {
  details?: string;
  state?: string;
};

export const DiscordRPC = {
  setActivity: (input: DiscordRpcActivityInput) =>
    callCommand<void>("discord_rpc_set_activity", {
      input: {
        clientId: DISCORD_CLIENT_ID,
        ...input,
      },
    }),
  clear: () => callCommand<void>("discord_rpc_clear"),
};

File: src-tauri/src/commands/discord.rs
use crate::{discord_rpc::DiscordRpcActivityInput, errors::AppResult, state::AppState};

#[tauri::command]
pub async fn discord_rpc_set_activity(
    state: tauri::State<'_, AppState>,
    input: DiscordRpcActivityInput,
) -> AppResult<()> {
    state.discord_rpc.set_activity(input)
}

#[tauri::command]
pub async fn discord_rpc_clear(state: tauri::State<'_, AppState>) -> AppResult<()> {
    state.discord_rpc.clear()
}
